FirebaseCore/Sources/FIRLogger.m

```diff
@@ -157,27 +157,44 @@ void FIRLogBasic(FIRLoggerLevel level,
 FIR_LOGGING_FUNCTION(Info)
 FIR_LOGGING_FUNCTION(Debug)
 
-// Swift does not support variadic function calls
-void FIRLogDebugSwift(FIRLoggerService service, NSString *messageCode, NSString *message) {
-  FIRLogDebug(service, messageCode, @"%@", message);
-}
-
-void FIRLogWarningSwift(FIRLoggerService service, NSString *messageCode, NSString *message) {
-  FIRLogWarning(service, messageCode, @"%@", message);
-}
-
 #undef FIR_MAKE_LOGGER
 
-#pragma mark - FIRLoggerWrapper
+#pragma mark - FIRLogger
+
+@implementation FIRLogger
 
-@implementation FIRLoggerWrapper
++ (void)logWithLevel:(FIRLoggerLevel)level
+             service:(FIRLoggerService)service
+                code:(NSString *)code
+             message:(NSString *)message
+                args:(va_list)args {
+  FIRLogBasic(level, service, code, message, args);
+}
 
 + (void)logWithLevel:(FIRLoggerLevel)level
-         withService:(FIRLoggerService)service
-            withCode:(NSString *)messageCode
-         withMessage:(NSString *)message
-            withArgs:(va_list)args {
-  FIRLogBasic(level, service, messageCode, message, args);
+             service:(FIRLoggerService)service
+                code:(NSString *)code
+             message:(NSString *)message {
+  if (level == FIRLoggerLevelMin) {
```

Comment:
This implementation could be cleaned up if it just called `FIRLogBasic` but that would use the GULs logger implementation.

The `FIRLog*Swift` functions called into the macro function defined in FIRLogger.m.

Reply:
Is there a reason we use the GUL implementation vs: https://github.com/firebase/firebase-ios-sdk/blob/e236acacafffa30de4537c9a7a3028e16f29b9a5/FirebaseCore/Sources/FIRLogger.m#L138-L158

Reply:
Maybe because FIRLogger was intended to be a wrapper over GULLogger?

Reply:
It looks like the generated macros end up calling `FIRLogBasic` anyway too.

Reply:
@paulb777, so you are OK with the current implementation or you think I should remove the above macro and just log through GULs?

Reply:
> It looks like the generated macros end up calling `FIRLogBasic` anyway too.

@andrewheard, I see it now. I'm going to update my new API's implementation to call directly to `FIRLogBasic`

Reply:
Thanks, I prefer that. Done in 4feb8fc071d23a004f55d9ad619022d071595e33